consensus/src/alerts/service.rs

```diff
@@ -0,0 +1,209 @@
+use crate::{
+    alerts::{
+        handler::Handler, Alert, AlertMessage, AlerterResponse, ForkingNotification, NetworkMessage,
+    },
+    Data, Hasher, MultiKeychain, Multisigned, NodeCount, Receiver, Recipient, Sender, Terminator,
+};
+use aleph_bft_rmc::{DoublingDelayScheduler, Message as RmcMessage, ReliableMulticast};
+use futures::{channel::mpsc, FutureExt, StreamExt};
+use log::{debug, error, warn};
+use std::time;
+
+const LOG_TARGET: &str = "AlephBFT-alerter";
+
+pub struct Service<H: Hasher, D: Data, MK: MultiKeychain> {
+    messages_for_network: Sender<(NetworkMessage<H, D, MK>, Recipient)>,
+    messages_from_network: Receiver<NetworkMessage<H, D, MK>>,
+    notifications_for_units: Sender<ForkingNotification<H, D, MK::Signature>>,
+    alerts_from_units: Receiver<Alert<H, D, MK::Signature>>,
+    rmc: ReliableMulticast<H::Hash, MK>,
+    messages_for_rmc: Sender<RmcMessage<H::Hash, MK::Signature, MK::PartialMultisignature>>,
+    messages_from_rmc: Receiver<RmcMessage<H::Hash, MK::Signature, MK::PartialMultisignature>>,
+    keychain: MK,
```

Comment:
You actually only use the index, so maybe just save that?

Reply:
fax